src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php

```diff
@@ -243,33 +210,36 @@ public function __destruct()
             $this->clonesCount = 0;
             $this->isCollected = true;
 
-            $h = headers_list();
-            $i = count($h);
-            array_unshift($h, 'Content-Type: '.ini_get('default_mimetype'));
-            while (0 !== stripos($h[$i], 'Content-Type:')) {
-                --$i;
-            }
+            // If the server dumper is set and server is up, dump were already sent to it.
+            if (!$this->serverDumper || !$this->serverDumper->isServerUp()) {
```

Comment:
talking my previous comments into account, not sure we need this "if": this code handles a very special situation, namely errors that break correct collection of data. Not a big issue if we keep the fallback logic here.